### DISCUSSION TITLE
Traffic under Insight of repository

### DISCUSSION THREAD

**Discussion by AnimishY** [community/community#23048] [New to GitHub]
<p>I created a website in HTML, and published it using GitHub Pages.</p>
<p>Now i regularly go to see traffic on my repository, and I have multiple questions about it.</p>
<ol>
<li>
<p>Can I use the views under traffic as count of people viewed my repository or the website?</p>
</li>
<li>
<p>My visit to my repository also count as a visit?</p>
</li>
<li>
<p>What’s the difference between unique visitors and visitors?</p>
</li>
<li>
<p>Who are cloners?</p>
</li>
<li>
<p>What is the difference between unique cloners and cloners?</p>
</li>
<li>
<p>Is there any way to know how many visitors visited my website?</p>
</li>
</ol>
<p>Thanks!!</p>
<p>First topic, hoping for quick help!!</p>

**Comment by ghost** [comment, marked as answer]
<p>Hi <a class="mention" href="/animishy">@AnimishY</a>! Welcome to the GitHub Support community!</p>
<p>The insights tab within your repository displays traffic for your repo, not your GitHub pages site itself.</p>
<p>If you are logged in to GitHub then your own page views won’t be counted.</p>
<p>Visitors is the total visits to your repo. Unique visitors refers to the number of individual users who have visited your repository and these are only counted once, regardless of how many times they visit the repo.</p>
<p>Cloners refers to people to have cloned or downloaded a copy of your code from GitHub. Again, unique cloners is the number if individual users who have cloned the repo, whereas cloners is the total number of clones.</p>
<p>If you want to track traffic to your actual pages site, the best way to do this is through configuring a third party analytics service, such as Google Analytics or Clicky.</p>
<p>I hope this helps! Let me know if you have any other questions!</p>

**Reply by gavrie01** [reply]
Hi! Thank you for the question and answer. Can I as a quick question as well please? 

"If you are logged in to GitHub then your own page views won’t be counted" - I am logged in to https://github.com/gavrie01/my_repo in WEB, this is "if you are logged in to GitHub" part of the above statement, right? 
When I navigate to Insights -> Traffic within repo, I see there 28 unique cloners, which I assume are different accounts on GitHub? At the same time the lower graph shows 1 or as of today 3 (?) unique users, one of which is obviously me, because I know which pages I visit.

Why there is always me, as 1 unique user, 2 or 3, but not 28? Am I doing something wrong mwybe?


I am sorry in advance for the question and for taking your time, but maybe someone can explain me "in very simple words"? to me it looks like ths tab does not work.

Here is the screenshot of my page:
![Screenshot 2024-01-03 220637](https://github.com/community/community/assets/95389076/10a0ddfc-70e4-45fe-beb8-fd190333942f)

**Comment by AnimishY** [comment]
<p>Hey <a class="mention" href="/robertnorrie">@robertnorrie</a>,</p>
<p>Thanks for reply!<br>
Concepts more clear now!</p>

**Comment by vineettalashi** [comment]
<p>Is Unique visitors counted based on device IP Address ? means if  i visit one of my repository (without logging in github) from 2 devices (laptop, mobile) , then unique visitors will be 2 ?<br>
<a class="mention" href="/robertnorrie">@robertnorrie</a></p>

**Reply by Aadit-Garg** [reply]
no not actually, because it doesn't calculates views from users who are not logged in

**Comment by dev-hd11** [comment]
I'm little confused that why is the number of visitors less than the cloners. Also how will I verify that the cloners are not publishing my code under thier name. And also why is that that people have cloned my repo instead of downloading the release

**Reply by Inwerpsel** [reply]
Probably bots that use the GitHub API, or some other means that is not counted, and pull/clone every time there are new commits.

**Reply by sidney-eliot** [reply]
> I'm little confused that why is the number of visitors less than the cloners. Also how will I verify that the cloners are not publishing my code under thier name. And also why is that that people have cloned my repo instead of downloading the release

Don't worry if people clone your repo, it more so means that people care about what you made and want to archive it for themselves for if something where to happen to that repo. People sometimes change their mind and completely remove / private repos.

**Comment by iamtonmoy0** [comment]
I have a question who are the unique visitors are.Are they found the repo from github or the traffic come from other search engine

**Comment by TarilAurus** [comment]
**I think its a bit unclear still how traffic works.**
See my picture, how can i have 1 visitor with 33views and 1 visitor but 46 uniqe clones?
can someone explain 🙃
![github](https://github.com/community/community/assets/154546281/8937c5a7-1958-4b49-9b2e-e20241a7c7a1)

**Reply by HiM636d** [reply]
I think those are cloners who did not visit the page, but used the repository adress to clone it from an editor. So they did not enter the repo they just used git commands feom their editor to clone the repo using the repo's address.

**Reply by raumildhandhukia** [reply]
But how can they find my repository name to clone if they wont visit my github page?

**Reply by tpmst** [reply]
Ok i will explain it to you guys.

![image](https://github.com/user-attachments/assets/49d2af12-c968-487f-8706-3b37e2e6d134)

Clones refer to how often your repository has been cloned. This can be done without opening your repository by using the ```git clone``` command.

Who clones my repo: 
1. People who clone all public repositories to search for passwords or sensitive information.
2. People who collect data for statistical purposes.
3. People who want to use your published application.

The unique cloners count increases when someone who has never cloned your repository before does so. For example, if I clone your repo twice, the clones count will go up by two, but the unique cloners count will increase by only one.

The same applies to views. If someone new visits your repo, both the views and unique visitors counts will increase by one. If the person has already visited your repo before, only the views count will go up by one.

**Reply by mojje90** [reply]
That explains it for me. Thank you🙏

**Comment by CICD-pic-avr-github-admin** [comment]
Does clones count in the insights traffic tab include 'downloads as ZIP' as well?
